crates/bevy_ecs/src/system/query.rs

```diff
@@ -488,6 +488,33 @@ where
         }
     }
 
+    /// Gets the result of a single-result query.
+    ///
+    /// Assumes this query has only one result and panics if there are no or multiple results.
+    /// If you want to handle the error case yourself you can use the [`Self::try_single`] variant.
+    ///
+    /// # Example
+    ///
+    /// ```
+    ///  # use bevy_ecs::system::{Query, QuerySingleError};
+    ///  # use bevy_ecs::prelude::IntoSystem;
+    /// struct PlayerScore(i32);
+    /// fn player_scoring_system(query: Query<&PlayerScore>) {
```

Comment:
The player scoring system would probably access the player score mutably (and it should be a resource with the current scheme)

I'm struggling to come up with a realistic example where you need read only access to a singleton entity's components. The only place I've used it is collision detection - detecting when the player collides with an obstacle.

Reply:
That is, I wonder if we should just remove the read only variants, or at least make the mutable version the default.

Reply:
> I'm struggling to come up with a realistic example where you need read only access to a singleton entity's components.

There's multiple cases where I just needed the player's position - these cases certainly exist.

Reply:
I updated the example to use a Query<&Vec2, With<Player>> to represent a player with a position.

Reply:
> I'm struggling to come up with a realistic example where you need read only access to a singleton entity's components.

I've also run into this when trying to access a camera's projection to perform math using it.

Reply:
Yeah, I was considering using an example using a Camera because that's what I used the most, but I think a player position is simple and common enough to showcase this feature.